workers/unity/Packages/io.improbable.gdk.core/NetworkStats/NetworkStatisticsSystem.cs

```diff
@@ -0,0 +1,58 @@
+using System.Diagnostics;
+using Unity.Entities;
+using UnityEngine;
+using Debug = UnityEngine.Debug;
+
+namespace Improbable.Gdk.Core.NetworkStats
+{
+    [DisableAutoCreation]
+    [UpdateInGroup(typeof(SpatialOSReceiveGroup.InternalSpatialOSReceiveGroup))]
+    [UpdateBefore(typeof(SpatialOSReceiveSystem))]
+    public class NetworkStatisticsSystem : ComponentSystem
+    {
+        private const int DefaultBufferSize = 60;
+
+        private readonly NetStats netStats = new NetStats(DefaultBufferSize);
+        private readonly NetFrameStats lastIncomingData = NetFrameStats.Pool.Rent();
+        private readonly NetFrameStats lastOutgoingData = NetFrameStats.Pool.Rent();
+
+        private float lastFrameTime;
+
+#if !UNITY_EDITOR
```

Comment:
is there any case in which we would want to run this outside of the unity editor? e.g. to log the network statistics of a managed worker?

Reply:
In the future, possibly - but for now, we don't have the capabilities so we will limit it to Editor only